eel/eellib/eellib.go

```diff
@@ -63,64 +68,147 @@ func EELGetHandlersForEvent(ctx Context, event interface{}, eelHandlerFactory *H
 }
 
 // EELGetPublishers is similar to EELTransformEvent but return slice of publishers instead of slice of events.
-func EELGetPublishers(ctx Context, event interface{}, eelHandlerFactory *HandlerFactory) ([]EventPublisher, error) {
+func EELGetPublishers(ctx Context, event interface{}, eelHandlerFactory *HandlerFactory) ([]EventPublisher, []error) {
 	if Gctx == nil {
-		return nil, errors.New("must call EELInit first")
+		return nil, []error{errors.New("must call EELInit first")}
 	}
 	if ctx == nil {
-		return nil, errors.New("ctx cannot be nil")
+		return nil, []error{errors.New("ctx cannot be nil")}
 	}
+	ctx = ctx.SubContext()
+	ClearErrors(ctx)
 	doc, err := NewJDocFromInterface(event)
 	if err != nil {
-		return nil, err
+		return nil, []error{err}
 	}
 	publishers := make([]EventPublisher, 0)
 	eelMatchingHandlers := eelHandlerFactory.GetHandlersForEvent(ctx, doc)
 	for _, h := range eelMatchingHandlers {
 		p, err := h.ProcessEvent(ctx, doc)
 		if err != nil {
-			return nil, err
+			return nil, []error{err}
 		}
 		publishers = append(publishers, p...)
 	}
-	return publishers, nil
+	return publishers, GetErrors(ctx)
 }
 
-// EELTransformEvent transforms single event based on set of configuration handlers. Can yield multiple results.
-func EELTransformEvent(ctx Context, event interface{}, eelHandlerFactory *HandlerFactory) ([]interface{}, error) {
+// EELGetPublishersConcurrent is the concurrent version of EELGetPublishers. Useful when processing multiple
+// expensive transformations at the same time (e.g. making heavy use of curl() function calls).
+func EELGetPublishersConcurrent(ctx Context, event interface{}, eelHandlerFactory *HandlerFactory) ([]EventPublisher, []error) {
 	if Gctx == nil {
-		return nil, errors.New("must call EELInit first")
+		return nil, []error{errors.New("must call EELInit first")}
 	}
 	if ctx == nil {
-		return nil, errors.New("ctx cannot be nil")
+		return nil, []error{errors.New("ctx cannot be nil")}
 	}
-	publishers, err := EELGetPublishers(ctx, event, eelHandlerFactory)
+	ctx = ctx.SubContext()
+	ClearErrors(ctx)
+	doc, err := NewJDocFromInterface(event)
 	if err != nil {
-		return nil, err
+		return nil, []error{err}
+	}
+	eelMatchingHandlers := eelHandlerFactory.GetHandlersForEvent(ctx, doc)
+	var mtx sync.Mutex
+	var wg sync.WaitGroup
+	publishers := make([]EventPublisher, 0)
+	errs := make([]error, 0)
+	for _, h := range eelMatchingHandlers {
+		wg.Add(1)
+		go func() {
```

Comment:
This may cause problem.  Please see details here:
https://github.com/golang/go/wiki/CommonMistakes